components/scripts/lib/git.sh

```diff
@@ -24,8 +24,12 @@ git_clone_project() {
    fi
 
    rm -rf "${clone_dir:?}"
-   # shellcheck disable=SC2086  # we want $branch to expand into multiple arguments
-   git clone --depth=1 ${branch} "${git_repo}" "${clone_dir}" || die "ERROR: Unable to clone from ${git_repo}." 2
+
+   # shellcheck disable=SC2086  # we want $git_options and $branch to expand into multiple arguments
+   debug git clone --depth=1 ${git_options} ${branch} "${git_repo}" "${clone_dir}"
```

Comment:
I suggest we also add `--shallow-submodules` by default.

Reply:
+1. I was thinking this yesterday as well after I reviewed the `git clone --help` content.